### PR TITLE
Warehouse Scene Creation Guidelines

### PR DESCRIPTION
**Pull request by pawelbudziszewski**
- In the Ros2FleetRobotTemplate added documentation with guidelines describing how to create or modify the warehouse scene
- In the WarehouseAssets gem extended readme by a description of the WarehouseScene.prefab

This addresses #183

### CODE REVIEW

File: Gems/WarehouseAssets/README.md [hunk 1]
```diff
@@ -82,15 +82,37 @@ The `Warehouse_structural` folder contains prefabs comprising the warehouse buil
...
+- `Storage_markings` contains number boards on storage racks.
+- `Floor_markings` contains decal objects with floor lines.
+- `Movables` contains `Storage_on_wheels` prefabs.
+- `Floor_tire_traces` contains decal objects with tire traces.
+- `Storage_racks` contains storage racks arranged in 2 groups:
+  - `Unpacked` with `Storage_rack` prefabs
+  - `Packs` with `Storage_2x2_Pack` prefabs
+- `Rack_protectors` contains rack protection prefabs
+
+These objects are followed by instances warehouse structural assets:
```
Comment (adamdbrw) [inline Gems/WarehouseAssets/README.md:103]:
```suggestion
These objects are followed by instances of warehouse structural assets:
```

Comment (adamdbrw) [reply Gems/WarehouseAssets/README.md:103]: Please run grammar / spellcheck for the entire thing to catch some minor issues like this.

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Warehouse Scene Creation Guidelines
+
+This document describes the process of creating and modifying the warehouse scene which is the bases of the ROS2 Fleet Template. All the prefabs and assets used in this scene can be found in the [Warehouse Assets Gem](../../../Gems/WarehouseAssets). Before proceeding please take a look at [README](../../../Gems/WarehouseAssets/README.md) of this gem. You'll find there a list of all included objects and some useful tips.
```
Comment (adamdbrw) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:3]:
```suggestion
This document describes the process of creating and modifying the warehouse scene which is the basis of the ROS2 Fleet Template. All the prefabs and assets used in this scene can be found in the [Warehouse Assets Gem](../../../Gems/WarehouseAssets). Before proceeding please take a look at [README](../../../Gems/WarehouseAssets/README.md) of this Gem. You'll find a list of all included objects and some useful tips there.
```

Comment (adamdbrw) [reply Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:3]: Similarly, please check the docs.

File: Gems/WarehouseAssets/README.md [hunk 3]
```diff
@@ -82,15 +82,37 @@ The `Warehouse_structural` folder contains prefabs comprising the warehouse buil
...
-#### Creating Warehouse scene
+- `Global_illumination` contains lights imitating global illumination and `Reflection Probe` objects.
+- `Storage_markings` contains number boards on storage racks.
+- `Floor_markings` contains decal objects with floor lines.
+- `Movables` contains `Storage_on_wheels` prefabs.
+- `Floor_tire_traces` contains decal objects with tire traces.
+- `Storage_racks` contains storage racks arranged in 2 groups:
+  - `Unpacked` with `Storage_rack` prefabs
+  - `Packs` with `Storage_2x2_Pack` prefabs
+- `Rack_protectors` contains rack protection prefabs
```
Comment (adamdbrw) [inline Gems/WarehouseAssets/README.md:101]:
```suggestion
- `Rack_protectors` contains rack protection prefabs.
```

Comment (adamdbrw) [reply Gems/WarehouseAssets/README.md:101]: Consistency in punctuation - check more items on this and other lists.

**Review by adamdbrw — COMMENTED** [review]
Are there relevant sections of documentation of O3DE Editor content creation this README could link to?

**Comment by pawelbudziszewski** [comment]
Thank you for the review

> Are there relevant sections of documentation of O3DE Editor content creation this README could link to?

Added some more clarifications with links to the O3DE docs

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Warehouse Scene Creation Guidelines
+
+This document describes the process of creating and modifying the warehouse scene which is the basis of the ROS2 Fleet Template. All the prefabs and assets used in this scene can be found in the [Warehouse Assets Gem](../../../Gems/WarehouseAssets). Before proceeding please take a look at [README](../../../Gems/WarehouseAssets/README.md) of this Gem. You'll find a list of all included objects and some useful tips there.
```
Comment (pijaro) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:3]: Nitpick :smiley: 
```suggestion
This document describes the process of creating and modifying the warehouse scene which is the basis of the ROS 2 Fleet Template. All the prefabs and assets used in this scene can be found in the [Warehouse Assets Gem](../../../Gems/WarehouseAssets). Before proceeding please take a look at [README](../../../Gems/WarehouseAssets/README.md) of this Gem. You'll find a list of all included objects and some useful tips there.
```

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+You can also type WarehouseScene in the search field of the `Pick Prefab` dialog.
+
+> Note: to find out more about instantiating prefabs please refer to [this document](https://www.o3de.org/docs/learning-guide/tutorials/entities-and-prefabs/entity-and-prefab-basics/#instantiate-a-prefab).
+
+For best visual results it is advised to do following things:
+
+- The `WarehouseScene` prefab is equipped with all the necessary lights. Any global or environmental lightning may interfere with internal lights. In particular, if you are creating the scene from scratch using the default O3DE template, you should consider removing all child entities of the `Atom Default Environment/` except for the camera.
+- In the Tips section of the [Warehouse Assets Gem documentation](../../../Gems/WarehouseAssets/README.md) you'll find a description of the camera configuration which will help achieve better visual results.
+
+> Note: the ROS2 Fleet Template already contains a scene with the `WarehouseScene.prefab` and fully configured camera.
```
Comment (pijaro) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:22]:
```suggestion
> Note: the ROS 2 Fleet Template already contains a scene with the `WarehouseScene.prefab` and fully configured camera.
```

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+If you have the `WarehouseScene` prefab on your scene you can easily apply modifications. It is possible to alter virtually any object on the scene, remove it or add a new one. The structure of the `WarehouseScene` prefab contains a few levels of nested prefabs (please refer to [this document](../../../Gems/WarehouseAssets/README.md) for a description). For example, prefabs with boxes are located inside prefabs with storage racks, which are often located in storage packs (containing 4 racks). See the image below.
+
+![](images/WarehouseSceneSampleTree.png)
+
+You can alter objects on each level by removing, scaling or moving any objects. For example, you can:
+
+- Re-arrange boxes on shelves by translating and rotating existing ones, or adding new ones,
+- Remove boxes from a shelf to make places for different objects,
+- Remove the whole storage rack,
+- Translate racks to different locations.
```
Comment (pijaro) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:35]:
```suggestion
- re-arrange boxes on shelves by translating and rotating existing ones, or adding new ones,
- remove boxes from a shelf to make places for different objects,
- remove the whole storage rack,
- translate racks to different locations.
```

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+> Note: in O3DE there are 2 ways you can alter instantiated prefab: by changing prefab instance or prefab definition (source). If you modify prefab source, all instances will be altered. Thus unless you really know what you are doing, you should edit instances only. Simple hint: do not double-click on the prefab in the Entity Outliner. When double-clicking the editor enters focused mode, prefab is highlighted in blue, changes will be saved in the prefab source. Please refer to [this document](https://www.o3de.org/docs/learning-guide/tutorials/entities-and-prefabs/entity-and-prefab-basics/) to learn about working with prefabs.
+
+## Creating the warehouse scene
+
+According to your needs, you may also create a custom warehouse using selected assets from the Warehouse Assets Gem. As described in the [Warehouse Assets Gem README](../../../Gems/WarehouseAssets/README.md), the gem contains structural assets defining the building and various objects to be placed inside. Structural assets include 3 prefabs:
+
+- Floor,
+- Walls with gates,
+- Roof with lights.
```
Comment (pijaro) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:51]:
```suggestion
- floor,
- walls with gates,
- roof with lights.
```

File: Templates/Ros2FleetRobotTemplate/docs/scene_creation.md [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+If you intend to use them together, be sure to place them with the same transform, to ensure alignment. Besides the building structure, the gem provides a set of different objects to be used in the warehouse. These are arranged into prefabs with different levels of nesting. You can import the empty rack and place boxes on it, but you can also instantiate ready-to-use prefab with racks and boxes or pack of 4 racks. Of course, nothing prevents you from modifying such prefabs after instantiating.
```
Comment (pijaro) [inline Templates/Ros2FleetRobotTemplate/docs/scene_creation.md:53]: > If you intend to use them together, be sure to place them with the same transform, to ensure alignment.

What do you mean here? Setting the same values to the transform component or putting them in the same parent?

**Review by michalpelka — APPROVED** [review]

**Review by adamdbrw — APPROVED** [review]

**Pull request merged**